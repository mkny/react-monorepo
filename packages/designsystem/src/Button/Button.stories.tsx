import { Meta } from "@storybook/react";

import { Button, Sizes } from "./Button";

// More on how to set up stories at: https://storybook.js.org/docs/7.0/react/writing-stories/introduction
const meta: Meta<typeof Button> = {
  title: "Example/Button",
  component: Button,
  tags: ["autodocs"],
  argTypes: {
    backgroundColor: { control: "color" },
  },
};

export default meta;

// More on writing stories with args: https://storybook.js.org/docs/7.0/react/writing-stories/args
export const Primary = {
  args: {
    primary: true,
    label: "Button",
  },
};

export const Secondary = {
  args: {
    label: "Button",
  },
};

export const Large = {
  args: {
    size: Sizes.large,
    label: "Button",
  },
};

export const Small = {
  args: {
    size: Sizes.small,
    label: "Button",
  },
};
