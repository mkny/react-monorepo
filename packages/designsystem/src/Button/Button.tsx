type Props = {
  label: string;
};

/**
 *
 */
function Button(props: Props) {
  const { label } = props;
  return <div>{label}</div>;
}

export default Button;
export { Button };
