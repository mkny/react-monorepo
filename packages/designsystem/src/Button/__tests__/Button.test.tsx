import { render, screen } from "@testing-library/react";

import { Button } from "../Button";

describe("<Button />", () => {
  const defaultProps = { label: "Button" };
  test("render", () => {
    const { asFragment } = render(<Button {...defaultProps} />);

    expect(asFragment()).toMatchSnapshot();
  });
  it("should render as primary state", () => {
    render(<Button {...defaultProps} primary />);

    const btn = screen.getByRole("button");

    expect(btn).toHaveClass("storybook-button--primary");
  });
  it("should change backgroundColor", () => {
    render(<Button {...defaultProps} backgroundColor="red" />);

    const btn = screen.getByRole("button");

    expect(btn).toHaveStyle({ backgroundColor: "red" });
  });
});
