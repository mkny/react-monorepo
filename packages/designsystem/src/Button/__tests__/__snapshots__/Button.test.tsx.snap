// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> render 1`] = `
<DocumentFragment>
  <button
    class="storybook-button storybook-button--medium storybook-button--secondary"
    type="button"
  >
    Button
  </button>
</DocumentFragment>
`;
